src/serac/physics/base_physics.hpp

```diff
@@ -339,18 +342,17 @@ class BasePhysics {
    * @brief Accessor for getting named finite element state primal solution from the physics modules at a given
    * checkpointed cycle index
    *
-   * @param state_name The name of the Finite Element State primal solution to retrieve
    * @param cycle The cycle to retrieve state from
    * @return The named primal Finite Element State
    */
-  virtual FiniteElementState loadCheckpointedState(const std::string& state_name, int cycle) const;
+  virtual std::unordered_map<std::string, FiniteElementState> getCheckpointedStates(int cycle) const;
```

Comment:
why is this signature changing?

Reply:
This should almost be a private function, but I think Kenny wanted it for error estimates, potentially.  Are you using this already Kenny?

Reply:
or maybe that was the get_adjoint function, sorry.

Reply:
Lido calls this method to grab the checkpointed states and pass them to our qoi operator (and access solutions on the sensitivity pass). This will be a breaking change that we will have to address in Lido, which is fine, just wanted to understand why/if it's necessary.

Reply:
There are two reasons:
1. Using the old disk-based method, this was causing the file to be opened once per field instead of a batch job. It's possible that this disk thrashing was part of our problem.
2. It makes the names slightly more consistent with other methods as this could just be reading from memory instead of loading it from disk.

I'm more concerned about 1, but either way should technically work.

Reply:
any thoughts @tupek2 @kswartz92 ? I think I've found the memory errors and I want to close this one out. I can either revert this change or leave it as-is.

Reply:
I am fairly indifferent.  I think consistency is good, so if we are typically returning maps, we should make that consistent throughout.  Sadly creates churn for Kenny if that is the case!

Reply:
I'd say I have a slight preference for returning the state. If the other concerns have been addressed, it seems more natural to not require the caller get every state at once. I'm not seeing many other places maps are returned

Reply:
I'll revert this change and we can revisit it once checkpointing by disk is more standard.